### ISSUE TITLE
CLI autostart crashes at login on Monterey due to unset UTF-8 locale?

### ISSUE THREAD

**Issue by duncanbrook** [samschott/maestral#475]
**Describe the bug**
CLI autostart crashes at login on Monterey, apparently due to unset UTF-8 locale?

**To Reproduce**

1. Turn off GUI start-on-login: this is about the CLI daemon.
2. `maestral autostart -Y`
3. Log out of MacOS & log in
4. See this window:
<img width="632" alt="Maestral error window" src="https://user-images.githubusercontent.com/7630013/137951995-0f9b4864-9674-413e-b400-d98be4febbd0.png">

Containing this error message:

>  Traceback (most recent call last):
>   File "Contents/Resources/app/maestral_cocoa/__main__.py", line 41, in <module>
>   File "Contents/Resources/app/maestral_cocoa/__main__.py", line 36, in freeze_support_cli
>   File "Contents/Resources/app_packages/click/core.py", line 1137, in __call__
>   File "Contents/Resources/app_packages/click/core.py", line 1043, in main
>   File "Contents/Resources/app_packages/click/_unicodefun.py", line 100, in _verify_python_env
> RuntimeError: Click will abort further execution because Python was configured to use ASCII as encoding for the environment. Consult https://click.palletsprojects.com/unicode-support/ for mitigation steps.
> 
> This system lists some UTF-8 supporting locales that you can pick from. The following suitable locales were discovered: af_ZA.UTF-8, am_ET.UTF-8, be_BY.UTF-8, bg_BG.UTF-8, ca_ES.UTF-8, cs_CZ.UTF-8, da_DK.UTF-8, de_AT.UTF-8, de_CH.UTF-8, de_DE.UTF-8, el_GR.UTF-8, en_AU.UTF-8, en_CA.UTF-8, en_GB.UTF-8, en_IE.UTF-8, en_NZ.UTF-8, en_US.UTF-8, es_ES.UTF-8, et_EE.UTF-8, eu_ES.UTF-8, fi_FI.UTF-8, fr_BE.UTF-8, fr_CA.UTF-8, fr_CH.UTF-8, fr_FR.UTF-8, he_IL.UTF-8, hr_HR.UTF-8, hu_HU.UTF-8, hy_AM.UTF-8, is_IS.UTF-8, it_CH.UTF-8, it_IT.UTF-8, ja_JP.UTF-8, kk_KZ.UTF-8, ko_KR.UTF-8, lt_LT.UTF-8, nl_BE.UTF-8, nl_NL.UTF-8, no_NO.UTF-8, pl_PL.UTF-8, pt_BR.UTF-8, pt_PT.UTF-8, ro_RO.UTF-8, ru_RU.UTF-8, sk_SK.UTF-8, sl_SI.UTF-8, sr_YU.UTF-8, sv_SE.UTF-8, tr_TR.UTF-8, uk_UA.UTF-8, zh_CN.UTF-8, zh_HK.UTF-8, zh_TW.UTF-8

5. `maestral status` returns "Maestral daemon is not running."
6. `maestral start` then works just fine.

**Expected behaviour**
When set to autostart, the maestral daemon should launch silently at login and run without errors.

**System:**
 - Maestral version: 1.5.0
 - Python version: Apple's bundled Python (python) is 2.7.18, & via Homebrew I also have Python 3 (python3) 3.9.7
 - OS: MacOS Monterey 12.0.1 (21A558)

**Additional context**

Following the advice at the link provided in the error text and elsewhere, I tried to get things working properly by setting

```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```
in my .zshrc (I use zsh) and in .bash_profile (just in case?)

But that did not help.

Aliasing python to python3 (apparently 3.9+ has changes that mitigate this issue?) also did not help, but I absolutely could be doing something wrong there.

I suspect that this issue may be Monterey-specific and am not sure where to go from here?

**Comment by SamSchott** [comment]
Oh, I haven't seen this before! This is also strange because the app environment should always have `LC_CTYPE=UTF-8` set.

Could you set the log level to DEBUG and check the logs when starting automatically after login? This should also print all environment variables to the log. Which locale-related env variables do you see there?

> in my .zshrc (I use zsh) and in .bash_profile (just in case?)

I am not sure if any environment settings from .zshrc are carried over to launch daemons. The launch script for Maestral is installed at `~/Library/LaunchAgens/com.samschott.maestral-daemon.maestral.plist` and you can modify it to directly set environment variables as described [here](https://serverfault.com/questions/111391/use-an-environment-variable-in-a-launchd-script).

**Comment by SamSchott** [comment]
I forgot to ask, are you using the app bundle or did you install the Python package from PyPI? Only the app bundle enforces `LC_CTYPE=UTF-8` in its plist, if you install the Python package you are currently responsible for locale settings yourself.

**Comment by duncanbrook** [comment]
Thanks for the replies!

I'm using the app bundle, but I accidentally clicked the "install" and then "uninstall" buttons for the CLI tools several times in a row (don't ask), so perhaps that not-ideal-user-behavior knocked something out of whack?

I followed your instructions to set the log level to debug and observed that level of logging working fine for an instance of maestral run from the terminal (i.e. `maestral log show` printed up lots of debug information).

When I logged out & logged back in to see what debug information made it to the log, I got the same "Application has crashed" window described in my first post, but `maestral log show` printed ... nothing.  The log is empty.

maestral runs just fine from the terminal, but I'm also seeing "Authentication failed | Please make sure that you entered the correct authentication code." errors, so ... before I mess with the provided plist, I'm going to uninstall everything maestral completely and try again fresh.

**Comment by duncanbrook** [comment]
1. Authentication errors appear to be a separate issue: https://github.com/SamSchott/maestral/issues/479
2. I removed everything maestral on my system, then did `brew install maestral` (in the previous go-round, I'd installed the app bundle by downloading it direct from Github).  The Homebrew cast installs both the app and the CLI tools.  I tried to configure entirely via the CLI and succeeded in linking/authenticating but encountered issues setting a directory location, so I fell back to running the GUI and doing that there.
3. After doing "Resume Syncing" a bunch of times (see #1), I successfully got my full Dropbox folder downloaded and again tried to test the CLI autostart by setting `maestral log level debug` and `maestral autostart -Y`
4. Again I get the "Application has crashed" window and `maestral log show` in the terminal shows no results from the attempted autostart session at all.

Now I will try editing the plist and see how that goes!
